extension/src/popup/components/WarningMessages/index.tsx

```diff
@@ -499,28 +500,34 @@ export const NewAssetWarning = ({
       .build()
       .toXDR();
 
-    const res = await dispatch(
-      signFreighterTransaction({
-        transactionXDR,
-        network: networkDetails.networkPassphrase,
-      }),
-    );
-
-    if (signFreighterTransaction.fulfilled.match(res)) {
-      const submitResp = await dispatch(
-        submitFreighterTransaction({
-          signedXDR: res.payload.signedTransaction,
-          networkDetails,
+    if (isHardwareWallet) {
+      await dispatch(startHwSign({ transactionXDR, shouldSubmit: true }));
+      emitMetric(METRIC_NAMES.manageAssetAddUnsafeAsset, { code, issuer });
```

Comment:
ran into this bug, where adding the trustline when the NewAssetWarning popup comes up w/ hardware wallet would throw error
(I fixed this in [this PR](https://github.com/stellar/freighter/pull/656) for the ScamAssetWarning popup, forgot to change here as well)